README.md
Ethereum Based Employee reward system

# [Employee Reward System](http://tnb.sriharikapu.com/system.png)
[![version][version-badge]][CHANGELOG]


[![Build Status](https://travis-ci.org/jibrelnetwork/ethereum-qr-code.svg?branch=master)](https://travis-ci.org/jibrelnetwork/ethereum-qr-code) ![DEPENDENCIES status](https://david-dm.org/jibrelnetwork/ethereum-qr-code.svg)

![](http://tnb.sriharikapu.com/system.png)

**[Employee Reward System](http://tnb.sriharikapu.com/system.png)** Reward systems are one of the most prominent and most important features of any organization. Because of the wide ranging impact a reward system has on individual behavior and behavior of organizations, it is very important designing such reward systems that will motivate the right performance, attract the right people and create a supportive and good working climate within organization.

As it was mentioned earlier, human resource management and reward systems play a significant role in this economic transformation. New recruits with appropriate personality profile can be the face of an economic transformation for which these youth need training. However, the employee of the former system cannot be forgotten and those who deserve have to receive promotion thus showing example for the new workers. As it indicates, human resource management is more than significant in the transformation of an economy where challenges "including lack of appropriate trained leaders, an education system that elevated technical ability above client facing skills, a shortage of linguists, marketers and financial professionals, and an inappropriate reward and remuneration system" (Rossides) can lead to potential pitfalls and organizational malfunctions.

Special thanks go to: TNB, Blockchaintmhub.io, UNITEN
Developers: #SrihariKapu, #Sandeep, #Bassam

## Coding Sources:

+ [AnimateCss](https://daneden.github.io/animate.css/)
+ [Bootstrap3](https://getbootstrap.com/)
+ [NginX](https://www.nginx.com/)
+ [Jquery](https://jquery.com/)
+ [Ethereum](https://ethereum.com/)
+ [MetaMask](https://metamask.io/)
+ [ParticleJS](https://vincentgarreau.com/particles.js/)
+ [Etherscan](https://etherscan.io/)
+ [GulpJS](https://gulpjs.com/)
+ [Django](https://www.djangoproject.com/)
+ [Python3](https://www.python.org)

## Quick Start:

Quick start options:

+ [Demo](https://breathing.sriahrikapu.com).

   User-Id  : 
   
   Password : 1234@qwerty 
   
for User-Id write to us : sri@sriharikapu.com

  
#### Credits to:

+ [Creative Tim](https://www.creative-tim.com/)


### What's included

Within the download you'll find the following directories and files:
```
Employee Reward System
├── Dashboard
│   ├── BS3
|   |   ├── assets
|   |   |   ├── css
|   |   |   |   ├── animate.min.css
|   |   |   |   ├── bootstrap.min.css
|   |   |   |   ├── demo.css
|   |   |   |   ├── leave_management.css
|   |   |   |   ├── light-bootstrap-dashboard.css
|   |   |   |   └── pe-icon-7-stroke.css
|   |   |   ├── fonts
|   |   |   |   ├── Pe-icon-7-stroke.eot
|   |   |   |   ├── Pe-icon-7-stroke.svg
|   |   |   |   ├── Pe-icon-7-stroke.ttf
|   |   |   |   └── Pe-icon-7-stroke.woff
|   |   |   ├── img
|   |   |   |   ├── bassam.jpg
|   |   |   |   ├── bassam1.jpg
|   |   |   |   ├── face-0.jpg
|   |   |   |   ├── face-1.jpg
|   |   |   |   ├── face-2.jpg
|   |   |   |   ├── face-3.jpg
|   |   |   |   ├── face-4.jpg
|   |   |   |   ├── face-5.jpg
|   |   |   |   ├── face-6.jpg
|   |   |   |   ├── face-7.jpg
|   |   |   |   ├── sandeep.jpg
|   |   |   |   ├── sri.jpg
|   |   |   |   └── Pe-icon-7-stroke.woff
|   |   |   ├── js
|   |   |   |   ├── bootstrap-notify.js
|   |   |   |   ├── bootstrap-select.js
|   |   |   |   ├── bootstrap.min.js
|   |   |   |   ├── chartist.min.js
|   |   |   |   ├── demo.js	
|   |   |   |   ├── jquery.3.2.1.min.js
|   |   |   |   ├── leave_management.js
|   |   |   |   ├── leave_management_backup-coffe-cafe.js
|   |   |   |   ├── light-bootstrap-dashboard.js
|   |   |   |   └── particle.min_flkakj.js
|   |   |   ├── sass
|   |   |   |   ├──lbd
|   |   |   |   |   └── mixins
|   |   |   |   |   |   ├── _buttons.scss
|   |   |   |   |   |   ├── _cards.scss
|   |   |   |   |   |   ├── _chartist.scss
|   |   |   |   |   |   ├── _icons.scss
|   |   |   |   |   |   ├── _inputs.scss
|   |   |   |   |   |   ├── _labels.scss
|   |   |   |   |   |   ├── _morphing-buttons.scss
|   |   |   |   |   |   ├── _navbars.scss
|   |   |   |   |   |   ├── _social-buttons.scss
|   |   |   |   |   |   ├── _tabs.scss
|   |   |   |   |   |   ├── _transparency.scss
|   |   |   |   |   |   └── _vendor-prefixes.scss
|   |   |   |   |   ├── _alerts.scss
|   |   |   |   |   ├── _buttons.scss
|   |   |   |   |   ├── _cards.scss
|   |   |   |   |   ├── _chartist.scss
|   |   |   |   |   ├── _checkbox-radio-switch.scss
|   |   |   |   |   ├── _dropdown.scss
|   |   |   |   |   ├── _footers.scss
|   |   |   |   |   ├── _inputs.scss
|   |   |   |   |   ├── _misc.scss
|   |   |   |   |   ├── _mixins.scss
|   |   |   |   |   ├── _navbars.scss
|   |   |   |   |   ├── _responsive.scss
|   |   |   |   |   ├── _sidebar-and-main-panel.scss
|   |   |   |   |   ├── _tables.scss
|   |   |   |   |   ├── _typography.scss
|   |   |   |   |   └── _variables.scss
|   |   |   |   └── light-bootstrap-dashboard.scss
|   |   |   └── test.html
|   |   ├── documentation
|   |   |   ├── css
|   |   |   |   └── documentation.css
|   |   |   └── tutorial-components.html
|   |   |
│   |   ├── CHANGELOG.md
│   |   ├── dashboard.html
│   |   ├── hr.html
│   |   ├── leave_management.html
│   |   ├── login.html
│   |   ├── market.html
│   |   ├── notifications.html
│   |   ├── table.html
│   |   ├── template.html
│   |   ├── typography.html
│   |   ├── upgrade.html
│   |   ├── user1.html
│   |   ├── user2.html
│   |   └── user3.html
│   ├── assets/js/
│   │   └── bootstrap.min.js
│   └── CHANGELOG.md   
├── Smartcontract
│   ├── tnb.sol
│   └── abi.json
├── TNB_HACKATHON/TNB
│   └──Django & PIP Dependencies
├── documents
│   ├── logo123.png
│   └── logo200x120.png
├── CODE_OF_CONDUCT.md
├── CONTRIBUTING.md
├── LICENSE
└── README.md

```

## Useful Links

Support us for More Dapps:

Ethereum: 0x42FE987767e39C9c2c19Dd24822f551e01791252

Bitcoin: 1HFGqnMP1cYvcaTqiMpqmr2gZRrJyhboYx

Ripple: rhwdjmxkuLbv4HoXe579N6pWobdF54u6Hz

Litecoin: LPxSWectcG5C7Dyp9EDTz7z4143pwqjeyo


## Find Us: 


Linkedin: <https://www.linkedin.com/in/bassam-abed-769227bb/>

Linkedin: <https://www.linkedin.com/in/sandeep-kewlani-4b002440/>

Linkedin: <https://www.linkedin.com/in/sriharikapu/>


## Social Media:

Twitter: <https://twitter.com/sriharikapu>

Facebook: <https://www.facebook.com/officialsriharikapu>

Google+: <https://plus.google.com/+sriharikapu>

Instagram: <https://instagram.com/sriharikapu>

[CHANGELOG]: ./CHANGELOG.md
[version-badge]: https://img.shields.io/badge/version-2.0.19.65-blue.svg



## Licenses

![Mozilla Public License 2.0](https://s3.amazonaws.com/cloud.ohloh.net/attachments/43318/mozilla_core_logo_med.jpg)

[![CC0](https://i.creativecommons.org/p/zero/1.0/88x31.png)](https://creativecommons.org/publicdomain/zero/1.0/)


Dashboard/CHANGELOG.md
# Change Log

## [2.0.1] 2017-12-12
### Bugfixing
- hide `.nav-mobile-menu` on sidebar
- navbar icons arrange
- other bug fixes

## [2.0.0] 2017-12-11
### Major Release
- update to Bootstrap 4 Beta
- added new icons by Nucleo
- bug fixes and improvements
- code refactoring

## [1.4.0] 2017-11-02
### Library Updates
- jQuery - `3.2.1`
- bootstrap - `3.3.7`
- chartist - `0.11.0`

### Bug Fixing
- replaced old javascript checkboxes and radios with only css checkboxes and radios
- fixed responsive sidebar
- small bug fixes

## [1.3.1] 2017-01-19
- switched to MIT license

## [1.3] 2016-01-22
### New Template page + Video Tutorial [current version]
- added the default template page + youtube video tutorial on how to create an Admin Template (link: https://www.youtube.com/watch...)

## [1.2] 2016-01-17
### New Page
- for those who want to upsell inside their dashboard we added a new page "Upgrade to PRO" with a pricing and options table

## [1.1] 2015-09-08
### Bug Fixing
- added company name/logo inside the sidebar for small screens
- fixed bug for notification with close button on small screens
- fix live preview bug for download on small screens
- fix table responsive for small screens
- added new labels for chartist on small screens

## [1.0] 2015-08-20
### Initial Release


Dashboard/BS3/assets/js/demo.js
type = ['','info','success','warning','danger'];


demo = {
    initPickColor: function(){
        $('.pick-class-label').click(function(){
            var new_class = $(this).attr('new-class');
            var old_class = $('#display-buttons').attr('data-class');
            var display_div = $('#display-buttons');
            if(display_div.length) {
            var display_buttons = display_div.find('.btn');
            display_buttons.removeClass(old_class);
            display_buttons.addClass(new_class);
            display_div.attr('data-class', new_class);
            }
        });
    },

    checkScrollForTransparentNavbar: debounce(function() {
            $navbar = $('.navbar[color-on-scroll]');
            scroll_distance = $navbar.attr('color-on-scroll') || 500;

            if($(document).scrollTop() > scroll_distance ) {
                if(transparent) {
                    transparent = false;
                    $('.navbar[color-on-scroll]').removeClass('navbar-transparent');
                    $('.navbar[color-on-scroll]').addClass('navbar-default');
                }
            } else {
                if( !transparent ) {
                    transparent = true;
                    $('.navbar[color-on-scroll]').addClass('navbar-transparent');
                    $('.navbar[color-on-scroll]').removeClass('navbar-default');
                }
            }
    }, 17),

    initDocChartist: function(){
        var dataSales = {
          labels: ['9:00AM', '12:00AM', '3:00PM', '6:00PM', '9:00PM', '12:00PM', '3:00AM', '6:00AM'],
          series: [
             [287, 385, 490, 492, 554, 586, 698, 695, 752, 788, 846, 944],
            [67, 152, 143, 240, 287, 335, 435, 437, 539, 542, 544, 647],
            [23, 113, 67, 108, 190, 239, 307, 308, 439, 410, 410, 509]
          ]
        };

        var optionsSales = {
          lineSmooth: false,
          low: 0,
          high: 800,
          showArea: true,
          height: "245px",
          axisX: {
            showGrid: false,
          },
          lineSmooth: Chartist.Interpolation.simple({
            divisor: 3
          }),
          showLine: false,
          showPoint: false,
        };

        var responsiveSales = [
          ['screen and (max-width: 640px)', {
            axisX: {
              labelInterpolationFnc: function (value) {
                return value[0];
              }
            }
          }]
        ];

        Chartist.Line('#chartHours', dataSales, optionsSales, responsiveSales);


        var data = {
          labels: ['Jan', 'Feb', 'Mar', 'Apr', 'Mai', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'],
          series: [
            [542, 443, 320, 780, 553, 453, 326, 434, 568, 610, 756, 895],
            [412, 243, 280, 580, 453, 353, 300, 364, 368, 410, 636, 695]
          ]
        };

        var options = {
            seriesBarDistance: 10,
            axisX: {
                showGrid: false
            },
            height: "245px"
        };

        var responsiveOptions = [
          ['screen and (max-width: 640px)', {
            seriesBarDistance: 5,
            axisX: {
              labelInterpolationFnc: function (value) {
                return value[0];
              }
            }
          }]
        ];

        Chartist.Bar('#chartActivity', data, options, responsiveOptions);

        var dataPreferences = {
            series: [
                [25, 30, 20, 25]
            ]
        };

        var optionsPreferences = {
            donut: true,
            donutWidth: 40,
            startAngle: 0,
            total: 100,
            showLabel: false,
            axisX: {
                showGrid: false
            }
        };

        Chartist.Pie('#chartPreferences', dataPreferences, optionsPreferences);

        Chartist.Pie('#chartPreferences', {
          labels: ['80%','10%','10%'],
          series: [80, 10, 10]
        });
    },

    initChartist: function(){

        var dataSales = {
          labels: ['9:00AM', '12:00AM', '3:00PM', '6:00PM', '9:00PM', '12:00PM', '3:00AM', '6:00AM'],
          series: [
             [287, 385, 490, 492, 554, 586, 698, 695, 752, 788, 846, 944],
            [67, 152, 143, 240, 287, 335, 435, 437, 539, 542, 544, 647],
            [23, 113, 67, 108, 190, 239, 307, 308, 439, 410, 410, 509]
          ]
        };

        var optionsSales = {
          lineSmooth: false,
          low: 0,
          high: 800,
          showArea: true,
          height: "245px",
          axisX: {
            showGrid: false,
          },
          lineSmooth: Chartist.Interpolation.simple({
            divisor: 3
          }),
          showLine: false,
          showPoint: false,
        };

        var responsiveSales = [
          ['screen and (max-width: 640px)', {
            axisX: {
              labelInterpolationFnc: function (value) {
                return value[0];
              }
            }
          }]
        ];

        Chartist.Line('#chartHours', dataSales, optionsSales, responsiveSales);


        var data = {
          labels: ['Jan', 'Feb', 'Mar', 'Apr', 'Mai', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'],
          series: [
            [542, 443, 320, 780, 553, 453, 326, 434, 568, 610, 756, 895],
            [412, 243, 280, 580, 453, 353, 300, 364, 368, 410, 636, 695]
          ]
        };

        var options = {
            seriesBarDistance: 10,
            axisX: {
                showGrid: false
            },
            height: "245px"
        };

        var responsiveOptions = [
          ['screen and (max-width: 640px)', {
            seriesBarDistance: 5,
            axisX: {
              labelInterpolationFnc: function (value) {
                return value[0];
              }
            }
          }]
        ];

        var data = {
          labels: ['13 Mar', '14 Mar', '15 Mar', '16 Mar', '18 Mar', '19 Mar', '20 Mar', '21 Mar', '22 Mar', '23 Mar', '24 Mar', '26 Mar'],
          series: [
              [7, 8, 7, 7, 8, 7, 8, 7, 7, 8, 8, 6],
              [8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8]
          ]
      };

        Chartist.Bar('#chartActivity', data, options, responsiveOptions);
        
        var dataPreferences = {
            series: [
                [25, 30, 20, 25]
            ]
        };

        var optionsPreferences = {
            donut: true,
            donutWidth: 40,
            startAngle: 0,
            total: 100,
            showLabel: false,
            axisX: {
                showGrid: false
            }
        };

        Chartist.Pie('#chartPreferences', dataPreferences, optionsPreferences);

        Chartist.Pie('#chartPreferences', {
          labels: ['80%','10%','10%'],
          series: [80, 10, 10]
        });
    },

    initGoogleMaps: function(){
        var myLatlng = new google.maps.LatLng(40.748817, -73.985428);
        var mapOptions = {
          zoom: 13,
          center: myLatlng,
          scrollwheel: false, //we disable de scroll over the map, it is a really annoing when you scroll through page
          styles: [{"featureType":"water","stylers":[{"saturation":43},{"lightness":-11},{"hue":"#0088ff"}]},{"featureType":"road","elementType":"geometry.fill","stylers":[{"hue":"#ff0000"},{"saturation":-100},{"lightness":99}]},{"featureType":"road","elementType":"geometry.stroke","stylers":[{"color":"#808080"},{"lightness":54}]},{"featureType":"landscape.man_made","elementType":"geometry.fill","stylers":[{"color":"#ece2d9"}]},{"featureType":"poi.park","elementType":"geometry.fill","stylers":[{"color":"#ccdca1"}]},{"featureType":"road","elementType":"labels.text.fill","stylers":[{"color":"#767676"}]},{"featureType":"road","elementType":"labels.text.stroke","stylers":[{"color":"#ffffff"}]},{"featureType":"poi","stylers":[{"visibility":"off"}]},{"featureType":"landscape.natural","elementType":"geometry.fill","stylers":[{"visibility":"on"},{"color":"#b8cb93"}]},{"featureType":"poi.park","stylers":[{"visibility":"on"}]},{"featureType":"poi.sports_complex","stylers":[{"visibility":"on"}]},{"featureType":"poi.medical","stylers":[{"visibility":"on"}]},{"featureType":"poi.business","stylers":[{"visibility":"simplified"}]}]

        }
        var map = new google.maps.Map(document.getElementById("map"), mapOptions);

        var marker = new google.maps.Marker({
            position: myLatlng,
            title:"Hello World!"
        });

        // To add the marker to the map, call setMap();
        marker.setMap(map);
    },

	showNotification: function(from, align){
    	color = Math.floor((Math.random() * 4) + 1);

    	$.notify({
        	icon: "pe-7s-gift",
        	message: "Welcome Back <b>Bassam Abed</b>"

        },{
            type: type[color],
            timer: 4000,
            placement: {
                from: from,
                align: align
            }
        });
	}


}

$.getScript("https://cdnjs.cloudflare.com/ajax/libs/particles.js/2.0.0/particles.min.js", function(){
    particlesJS('particles-js',
      {
        "particles": {
          "number": {
            "value": 80,
            "density": {
              "enable": true,
              "value_area": 800
            }
          },
          "color": {
            "value": "#ffffff"
          },
          "shape": {
            "type": "circle",
            "stroke": {
              "width": 0,
              "color": "#000000"
            },
            "polygon": {
              "nb_sides": 5
            },
            "image": {
              "width": 100,
              "height": 100
            }
          },
          "opacity": {
            "value": 0.5,
            "random": false,
            "anim": {
              "enable": false,
              "speed": 1,
              "opacity_min": 0.1,
              "sync": false
            }
          },
          "size": {
            "value": 5,
            "random": true,
            "anim": {
              "enable": false,
              "speed": 40,
              "size_min": 0.1,
              "sync": false
            }
          },
          "line_linked": {
            "enable": true,
            "distance": 150,
            "color": "#ffffff",
            "opacity": 0.4,
            "width": 1
          },
          "move": {
            "enable": true,
            "speed": 6,
            "direction": "none",
            "random": false,
            "straight": false,
            "out_mode": "out",
            "attract": {
              "enable": false,
              "rotateX": 600,
              "rotateY": 1200
            }
          }
        },
        "interactivity": {
          "detect_on": "canvas",
          "events": {
            "onhover": {
              "enable": true,
              "mode": "repulse"
            },
            "onclick": {
              "enable": true,
              "mode": "push"
            },
            "resize": true
          },
          "modes": {
            "grab": {
              "distance": 400,
              "line_linked": {
                "opacity": 1
              }
            },
            "bubble": {
              "distance": 400,
              "size": 40,
              "duration": 2,
              "opacity": 8,
              "speed": 3
            },
            "repulse": {
              "distance": 200
            },
            "push": {
              "particles_nb": 4
            },
            "remove": {
              "particles_nb": 2
            }
          }
        },
        "retina_detect": true,
        "config_demo": {
          "hide_card": false,
          "background_color": "#b61924",
          "background_image": "",
          "background_position": "50% 50%",
          "background_repeat": "no-repeat",
          "background_size": "cover"
        }
      }
    );

});